src/main/cpp/lang/js.cpp

```diff
@@ -1615,10 +1615,102 @@ struct [[nodiscard]] Highlighter : Highlighter_Base {
         if (number.erroneous) {
             emit_and_advance(number.length, Highlight_Type::error);
```

Comment:
I would assume there needs to be a `return true;` here. If the number is in a bad format and you've already emitted the highlight, there's nothing more to be done in this function.